api4/team.go

```diff
@@ -95,6 +95,29 @@ func createTeam(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Freemium enabled, on a cloud license. We must check limits before allowing to restore
+	if c.App.Config().FeatureFlags != nil && c.App.Config().FeatureFlags.CloudFree && (c.App.Channels().License() != nil && c.App.Channels().License().Features != nil && *c.App.Channels().License().Features.Cloud) {
```

Comment:
The feature flag check doesn't hurt but in theory if Freemium is not enabled limits are empty so it will just not guard below

Reply:
Going to leave it for explicitness - mainly so that the code gets double checked when the feature flag is removed down the line